pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -447,20 +687,17 @@ func (r *ReconcileWildFlyServer) statefulSetForWildFly(w *wildflyv1alpha1.WildFl
 // loadBalancerForWildFly returns a loadBalancer service
 func (r *ReconcileWildFlyServer) loadBalancerForWildFly(w *wildflyv1alpha1.WildFlyServer) *corev1.Service {
 	labels := labelsForWildFly(w)
-	sessionAffinity := corev1.ServiceAffinityNone
-	if w.Spec.SessionAffinity {
-		sessionAffinity = corev1.ServiceAffinityClientIP
-	}
 	loadBalancer := &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      loadBalancerServiceName(w),
 			Namespace: w.Namespace,
 			Labels:    labels,
 		},
 		Spec: corev1.ServiceSpec{
-			Type:            corev1.ServiceTypeLoadBalancer,
-			Selector:        labels,
-			SessionAffinity: sessionAffinity,
+			Type:     corev1.ServiceTypeClusterIP,
```

Comment:
Why are you changing the service type?

Reply:
Yeap, this is one thing I wanted to talk about with you.

For the ejb remote communication and the transaction recovery works correctly the service needs to be headless (https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
The reason for that is that headless service does not process loadbalancing. The headless service does only DNS resolution. The loadbalancing is the managed by EJB client. The reason for that is that JEB connection and recovery needs the stable connection endpoint. It can't be loadbalanced to arbitrary node in the cluster. The recovery re-connection has to end up at the same node where EJB bussiness logic was processed.

But as I'm writing this I found that I didn't realize that this change influences the other connection processing (e.g. HTTP) which benefits from the load balancing functionality. I think the solution could be to create a new service which would be headless and which would be used for server to server ejb communication. The other service (the original one) would be used for access of other protocols. I need to verify if it's feasible to do so but I think it should work.
What do you think?

Reply:
Server-to-server EJB communication are going through HTTP upgrade, right?

Reply:
Yes, the EJB communication goes over the HTTP upgrade but there is possible to run EJB communication only over HTTP (https://github.com/wildfly/wildfly-http-client).

I'm not sure how many users used that option but at least no one with the transaction context propagation as it was not working (https://issues.jboss.org/browse/WFLY-12184)